lib/miq_automation_engine/service_models/miq_ae_service_container_deployment.rb

```diff
@@ -0,0 +1,181 @@
+          @object.customizations[:agent][key] = val
+        end
+        @object.save!
+      end
+    end
+
+    def perform_scp(ip, username, local_path, remote_path)
+      Net::SCP.upload!(ip,
+                       username,
+                       local_path,
+                       remote_path,
+                       :ssh => {:key_data => ssh_auth.auth_key})
+    end
+
+    def perform_agent_commands(ip, username, commands = [])
+      ssh = LinuxAdmin::SSH.new(ip, username, ssh_auth.auth_key)
+      ssh.perform_commands(commands)
+    end
+
+    def check_connection(deployment_host_ip, username, nodes_ips)
+      agent = LinuxAdmin::SSHAgent.new(ssh_auth.auth_key, "/tmp/ssh_manageiq/ssh_manageiq_#{id}")
+      success, unreachable_ips = agent.with_service do |socket|
+        ssh = LinuxAdmin::SSH.new(deployment_host_ip, username)
+        unreachable_ips = nodes_ips.select do |sub_ip|
+          !ssh.perform_commands(["sudo -E ssh -o 'StrictHostKeyChecking no' #{username}@#{sub_ip} echo $?"], socket)[:stdout].include?("0\r\n")
+        end
+        [unreachable_ips.empty?, unreachable_ips]
+      end
+      unless success
+        raise StandardError, "couldn't connect to : #{unreachable_ips.join(',')}"
+      end
+      success
+    end
+
+    def playbook_running?
+      pid = ar_method do
+        @object.customizations[:agent][:ansible_pid] if @object.customizations[:agent]
+      end
+      !pid.nil?
+    end
+
+    def run_playbook_command(ip, username, cmd)
+      result = {:finished => false}
+      ansible_pid = ar_method do
+        @object.customizations[:agent][:ansible_pid] if @object.customizations[:agent]
+      end
+      ssh = LinuxAdmin::SSH.new(ip, username, ssh_auth.auth_key)
+      if ansible_pid
+        process_running = ssh.perform_commands(["sudo kill -0 #{ansible_pid}"])[:stdout]
+        unless process_running.empty?
+          result[:finished] = true
+          result[:stdout] = ssh.perform_commands(["sudo cat /tmp/ansible.log"])[:stdout]
+          stop_agent(ssh)
+        end
+      else
+        pid, socket = create_agent(ssh, username)
+        ssh.perform_commands(["sudo rm -f /tmp/*ansible.log*"])
+        ssh.perform_commands(["SSH_AUTH_SOCK=#{socket} nohup sudo -b -E #{cmd}"])
+        ansible_pid = ssh.perform_commands(["sudo pgrep -f 'sudo -b -E'"])[:stdout].split("\r").first
+        customize(:ansible_pid => ansible_pid, :agent_pid => pid, :agent_socket => socket)
+      end
+      result
+    end
+
+    def agent_exists?
+      ar_method do
+        @object.customizations[:agent][:agent_pid] if @object.customizations[:agent]
+      end
+    end
+
+    def create_agent(ssh, username)
+      output = ssh.perform_commands(["ssh-agent -a ssh_manageiq_#{id}"])[:stdout]
+      socket = output.split('=')[1].split(' ')[0].chop
+      socket = "/#{username}/#{socket}"
+      socket = "/home#{socket}" if username != "root"
+      pid = output.split('=')[2].split(' ')[0].chop
```

Comment:
>
> Here's the output, if by your comment you meant to add a command on the
> code that's something I rather not doing.


IMHO that would help the next person to read this code.
Or maybe something like this would be more self-documenting:

vars = {}
output.scan(/^(\w+)=([^;\n]+)/) { |k, v| vars[k] = v }
socket, pid = vars.values_at('SSH_AUTH_SOCK', 'SSH_AGENT_PID')